powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/repository/OperationRepository.java

```diff
@@ -61,4 +64,15 @@
             "ORDER BY o.timestampCreated")
     Stream<OperationEntity> findExpiredPendingOperations(Date timestamp);
 
+    @Query("SELECT o.id FROM OperationEntity o " +
+            "WHERE o.timestampExpires < :timestamp AND o.status = io.getlime.security.powerauth.app.server.database.model.OperationStatusDo.PENDING ")
+    List<String> findExpiredPendingOperationIds(Date timestamp);
+
+    @Modifying( clearAutomatically = true)
+    @Query("UPDATE OperationEntity o SET status = io.getlime.security.powerauth.app.server.database.model.OperationStatusDo.EXPIRED " +
```

Comment:
Please ensure that only operations with the status `PENDING` are changed to `EXPIRED`